src/test/java/reproductions/SubscriptionReproduction.java

```diff
@@ -0,0 +1,202 @@
+package reproductions;
+
+import graphql.ExecutionInput;
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import graphql.execution.SubscriptionExecutionStrategy;
+import graphql.schema.DataFetchingEnvironment;
+import graphql.schema.GraphQLSchema;
+import graphql.schema.idl.RuntimeWiring;
+import graphql.schema.idl.SchemaGenerator;
+import graphql.schema.idl.SchemaParser;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Supplier;
+
+import static graphql.schema.idl.RuntimeWiring.newRuntimeWiring;
+import static graphql.schema.idl.TypeRuntimeWiring.newTypeWiring;
+
+/**
+ * Related to <a href="https://github.com/spring-projects/spring-graphql/issues/949">...</a>
+ * <p>
+ * This reproduction is to see what's happening with Subscriptions and whether they keep their
+ * order when values are async.
+ */
+public class SubscriptionReproduction {
```

Comment:
This is a `main()` reproduction class - not a test - I wanted to find out  what was happening and I thought this was a good way to find out.  Other code now tests that the fix is in place